ge0/ge0_tests/geo_url_tests.cpp

```diff
@@ -96,6 +96,30 @@ UNIT_TEST(GeoUrl_Geo)
   TEST_ALMOST_EQUAL_ABS(info.m_lat, 53.666, kEps, ());
   TEST_ALMOST_EQUAL_ABS(info.m_lon, 0.0, kEps, ());
 
+  // URL Encoded comma (%2C) as delimiter
+  TEST(parser.Parse("geo:-18.9151863%2C-48.28712359999999?q=-18.9151863%2C-48.28712359999999", info), ());
+  TEST(info.IsLatLonValid(), ());
+  TEST_ALMOST_EQUAL_ABS(info.m_lat, -18.9151863, kEps, ());
+  TEST_ALMOST_EQUAL_ABS(info.m_lon, -48.28712359999999, kEps, ());
+
+  // URL Encoded comma (%2C) and space (%20)
+  TEST(parser.Parse("geo:-18.9151863%2C%20-48.28712359999999?q=-18.9151863%2C-48.28712359999999", info), ());
+  TEST(info.IsLatLonValid(), ());
+  TEST_ALMOST_EQUAL_ABS(info.m_lat, -18.9151863, kEps, ());
+  TEST_ALMOST_EQUAL_ABS(info.m_lon, -48.28712359999999, kEps, ());
+
+  // URL Encoded comma (%2C) and space as +
+  TEST(parser.Parse("geo:-18.9151863%2C+-48.28712359999999?q=-18.9151863%2C-48.28712359999999", info), ());
```

Comment:
Complicating the RFC regex looks riskier and non-standard, and requires more CPU when parsing standard RFC-compliant addresses. An approach with a fallback to urldecode if parsing failed looks clearer. Does it make sense to add these tests into that other PR?

Reply:
Roger, will update the other PR.